monai/transforms/spatial/dictionary.py

```diff
@@ -698,7 +698,8 @@ def __call__(
             affine = self.rand_affine.rand_affine_grid.get_transformation_matrix()
         else:
             grid = create_grid(spatial_size=sp_size)
-            affine = torch.eye(len(sp_size) + 1)
+            # to be consistent with the self._do_transform case (dtype and device)
+            affine = torch.as_tensor(np.eye(len(sp_size) + 1), device=self.rand_affine.rand_affine_grid.device)
```

Comment:
sorry for the confusion, this one should fix #1837, the other change is for #1827